core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/join/methods/InnerJoin.java

```diff
@@ -16,23 +16,41 @@
 
 package uk.gov.gchq.gaffer.operation.impl.join.methods;
 
-import com.google.common.collect.ImmutableMap;
 
 import uk.gov.gchq.gaffer.operation.impl.join.match.Match;
-import uk.gov.gchq.gaffer.operation.impl.join.match.MatchKey;
+import uk.gov.gchq.koryphe.tuple.MapTuple;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class InnerJoin implements JoinFunction {
+/**
+ * {@code InnerJoin} is a join function which returns matched items from an iterable and list.
+ */
+public class InnerJoin extends JoinFunction {
     @Override
-    public List join(final List left, final List right, final Match match, final MatchKey matchKey) {
-        List resultList = new ArrayList<>();
-        if (matchKey.equals(MatchKey.LEFT)) {
-            left.stream().filter(listObj -> !match.matching(listObj, right).isEmpty()).forEach(listObj -> resultList.add(ImmutableMap.of(listObj, match.matching(listObj, right))));
-        } else if (matchKey.equals(MatchKey.RIGHT)) {
-            right.stream().filter(listObj -> !match.matching(listObj, left).isEmpty()).forEach(listObj -> resultList.add(ImmutableMap.of(listObj, match.matching(listObj, left))));
+    public List<MapTuple> join(final Iterable keys, final List matchCandidates, final String keyName, final String matchingValuesName, final Match match, final Boolean flatten) {
```

Comment:
Can we keep these methods protected.  They are protected in super class and public here.  Same for the rest.